src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/client/AssessRisksNeedsApiClient.kt

```diff
@@ -15,21 +20,27 @@ class AssessRisksNeedsApiClient(private val webClient: WebClient) {
     return webClient
       .get()
       .uri("/risks/crn/$crn/widget")
-      .retrieve()
-      .onStatus(
-        { httpStatus -> HttpStatus.NOT_FOUND == httpStatus },
-        { Mono.empty() }
-      ).awaitBodyOrNull()
+      .awaitExchangeOrNull { response ->
+        when (response.statusCode()) {
+          HttpStatus.OK -> response.awaitBody()
+          HttpStatus.NOT_FOUND -> null
+          else -> throw response.createExceptionAndAwait()
+        }
+      }
   }
 
   fun getRiskPredictors(crn: String): Flow<RiskPredictor> {
```

Comment:
This one doesn't have `suspend` but `AssessmentApiClient .getAssessment` does - but they both use `exchangeToFlow`?!

Reply:
I have no idea why but `exchangeToFlow` is not a suspending function so the code still compiles, I've changed it to suspend as it does look odd